x/ecocredit/server/marketplace/query_buy_orders_test.go

```diff
@@ -0,0 +1,51 @@
+package marketplace
+
+import (
+	"testing"
+
+	"gotest.tools/v3/assert"
+
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+
+	api "github.com/regen-network/regen-ledger/api/regen/ecocredit/marketplace/v1"
+	v1 "github.com/regen-network/regen-ledger/x/ecocredit/marketplace"
+)
+
+func TestQueryBuyOrders(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+	assert.NilError(t, s.marketStore.BuyOrderTable().Insert(s.ctx, &api.BuyOrder{Buyer: s.addr}))
+	assert.NilError(t, s.marketStore.BuyOrderTable().Insert(s.ctx, &api.BuyOrder{Buyer: s.addr}))
+
+	res, err := s.k.BuyOrders(s.ctx, &v1.QueryBuyOrdersRequest{})
+	assert.NilError(t, err)
+	assert.Equal(t, 2, len(res.BuyOrders))
+}
+
+func TestQueryBuyOrdersByAddress(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+	_, _, otherAddr := testdata.KeyTestPubAddr()
+	_, _, noOrdersAddr := testdata.KeyTestPubAddr()
```

Comment:
Maybe we should add more addresses to the suite setup?

Reply:
i would be open to refactoring the suite to have a slice of addresses, and specifying the amount to fill in `setupBase` in a follow-up PR